src/app/services/ConfigRoomService.ts

```diff
@@ -123,26 +123,30 @@ export class ConfigRoomService {
             return;
         } else {
             const configRoom: ConfigRoom = configRoomOpt.get();
+            const configRoomUpdates: Promise<void>[] =
+                [this.configRoomDAO.removeCandidate(this.observedConfigRoomId, user)];
             if (configRoom.chosenOpponent?.id === user.id) {
                 // if the chosenOpponent leave, we're back to initial part creation
                 const update: Partial<ConfigRoom> = {
                     chosenOpponent: null,
                     partStatus: PartStatus.PART_CREATED.value,
                 };
-                await this.configRoomDAO.update(this.observedConfigRoomId, update);
+                configRoomUpdates.push(this.configRoomDAO.update(this.observedConfigRoomId, update));
             }
-            await this.configRoomDAO.removeCandidate(this.observedConfigRoomId, user);
+            await Promise.all(configRoomUpdates);
         }
     }
     public async deleteConfigRoom(candidates: MinimalUser[]): Promise<void> {
         display(ConfigRoomService.VERBOSE,
                 'ConfigRoomService.deleteConfigRoom(); this.observedConfigRoomId = ' + this.observedConfigRoomId);
         assert(this.observedConfigRoomId != null, 'ConfigRoomService is not observing a configRoom');
         const configRoomId: string = Utils.getNonNullable(this.observedConfigRoomId);
-        await this.configRoomDAO.delete(configRoomId);
         for (const candidate of candidates) {
+            // Need to delete the candidates before the actual configRoom,
+            // for the security rules to check that we are allowed to do it
```

Comment:
nous avons besoin de supprimer les candidats de la configRoom actuelle, pour que les règles de sécurité checkent que nous somme authoriser de le faire ?

Y a une étrangeté de formulation non ? Si je fais pas A, sécu sais pas vérifier si j'ai le droit de faire A :D

Reply:
& PS: là aussi p-e créer une array de Promises, puis faire un Promise.all ?
Aussi, Firestore a pas un "delete full subcollection" ?

Reply:
Promise.all pas possible ici, il faut s'assurer que ça se fasse dans le bon ordre ! Et non pas de « delete full subcollection », la doc indique de supprimer à la main:

> To delete an entire collection or subcollection in Cloud Firestore, retrieve all the documents within the collection or subcollection and delete them.

Pour la formulation: le « it » fait référence à la suppression de la config room, mais j'explicite

Reply:
Il faut s'assurer que la personne définie par la configRoom supprime les candidats, ok, mais, ça empêche de Promise.all()er parce queee ?

Reply:
Parce que si tu supprime d'abord la configRoom avant les candidats, comment tu check que c'est « la personne définie par la configRoom » qui supprime les candidats ?

Reply:
Promise.all sur les candidats puis ensuite configRoom ?

Reply:
ah pour ça bien, fait !